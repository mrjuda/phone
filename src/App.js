import React from 'react';
import {
  BrowserRouter, Routes, Route,
} from 'react-router-dom';
import Navbar from './components/NavBar';
import Rockets from './components/rockets/Rockets';
import Brands from './components/brands/Brands';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Navbar />
        <Routes>
          <Route path="/" element={<Brands />} />
          <Route path="/rockets" element={<Rockets />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
