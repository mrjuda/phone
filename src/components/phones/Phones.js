// Phones.js

import React, { useEffect, useState } from 'react';
import { useParams, NavLink } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { getPhones } from '../../redux/phones/phonesAC';
import PhoneData from './PhoneData';
import '../../index.css';
import btnGoBack from '../icons/arrow-left.png';
import btnMic from '../icons/mic.png';
import btnSettings from '../icons/gear.png';

const Phones = () => {
  const { slug } = useParams();
  const phones = useSelector((store) => store.phones);
  const [brandName, setBrandName] = useState('');
  useEffect(() => {
    if (phones.length > 0) {
      setBrandName(phones[1].brandName);
    }
  });

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getPhones(slug));
  }, []);
  return (
    <div className="phones-container">
      <div className="header">
        <div className="h-c1">
          <NavLink className="nav-icon" to="/">
            <img src={btnGoBack} className="nav-icon" alt="logo" />
          </NavLink>
          <h2 className="nav-text">2022</h2>
        </div>
        <div className="h-c2">
          <h1 className="header-title">
            devices by
            {' '}
            {brandName}
          </h1>
        </div>
        <div className="h-c3">
          <img src={btnMic} className="nav-icon" alt="logo" />
          <img src={btnSettings} className="nav-icon" alt="logo" />
        </div>
      </div>
      <div className="banner">DEVICES LIST</div>
      <ul className="phones-list">
        {phones.map((phone) => (
          <PhoneData key={phone.phoneSlug} phoneData={phone} />
        ))}
      </ul>
    </div>
  );
};

export default Phones;
