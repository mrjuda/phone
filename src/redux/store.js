import { configureStore, combineReducers } from '@reduxjs/toolkit';
import brandsSlice from './brands/brandsSlice';
import phonesSlice from './phones/phonesSlice';

const reducer = combineReducers({
  brands: brandsSlice.reducer,
  phones: phonesSlice.reducer,
});

const store = configureStore({ reducer });

export default store;
