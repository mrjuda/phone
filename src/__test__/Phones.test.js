// Brands.test.js

import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Phones from '../components/phones/Phones';
import App from '../App';
import store from '../redux/store';

describe('Phones.js file rendering ', () => {
  test('Phones renders correctly', () => {
    const phones = render(
      <Provider store={store}>
        <App>
          <Phones />
        </App>
      </Provider>,
    );
    expect(phones).toMatchSnapshot();
  });

  it('Brand component renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App>
            <Phones />
          </App>
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
