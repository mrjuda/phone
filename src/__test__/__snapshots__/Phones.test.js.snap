// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brands.js file rendering  Brand component renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="brands-container"
  >
    <div
      className="header"
    >
      <div
        className="h-c1"
      >
        <img
          alt="logo"
          className="nav-icon"
          src="arrow-left.png"
        />
        <h2
          className="nav-text"
        >
          2022
        </h2>
      </div>
      <div
        className="h-c2"
      >
        <h1
          className="header-title"
        >
          devices by brand
        </h1>
      </div>
      <div
        className="h-c3"
      >
        <img
          alt="logo"
          className="nav-icon"
          src="mic.png"
        />
        <img
          alt="logo"
          className="nav-icon"
          src="gear.png"
        />
      </div>
    </div>
    <div
      className="banner"
    >
      DEVICES BY MAKER
    </div>
    <ul
      className="brands-list"
    />
  </div>
</div>
`;

exports[`Brands.js file rendering  Brands renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="brands-container"
        >
          <div
            class="header"
          >
            <div
              class="h-c1"
            >
              <img
                alt="logo"
                class="nav-icon"
                src="arrow-left.png"
              />
              <h2
                class="nav-text"
              >
                2022
              </h2>
            </div>
            <div
              class="h-c2"
            >
              <h1
                class="header-title"
              >
                devices by brand
              </h1>
            </div>
            <div
              class="h-c3"
            >
              <img
                alt="logo"
                class="nav-icon"
                src="mic.png"
              />
              <img
                alt="logo"
                class="nav-icon"
                src="gear.png"
              />
            </div>
          </div>
          <div
            class="banner"
          >
            DEVICES BY MAKER
          </div>
          <ul
            class="brands-list"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="brands-container"
      >
        <div
          class="header"
        >
          <div
            class="h-c1"
          >
            <img
              alt="logo"
              class="nav-icon"
              src="arrow-left.png"
            />
            <h2
              class="nav-text"
            >
              2022
            </h2>
          </div>
          <div
            class="h-c2"
          >
            <h1
              class="header-title"
            >
              devices by brand
            </h1>
          </div>
          <div
            class="h-c3"
          >
            <img
              alt="logo"
              class="nav-icon"
              src="mic.png"
            />
            <img
              alt="logo"
              class="nav-icon"
              src="gear.png"
            />
          </div>
        </div>
        <div
          class="banner"
        >
          DEVICES BY MAKER
        </div>
        <ul
          class="brands-list"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Phones.js file rendering  Brand component renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="brands-container"
  >
    <div
      className="header"
    >
      <div
        className="h-c1"
      >
        <img
          alt="logo"
          className="nav-icon"
          src="arrow-left.png"
        />
        <h2
          className="nav-text"
        >
          2022
        </h2>
      </div>
      <div
        className="h-c2"
      >
        <h1
          className="header-title"
        >
          devices by brand
        </h1>
      </div>
      <div
        className="h-c3"
      >
        <img
          alt="logo"
          className="nav-icon"
          src="mic.png"
        />
        <img
          alt="logo"
          className="nav-icon"
          src="gear.png"
        />
      </div>
    </div>
    <div
      className="banner"
    >
      DEVICES BY MAKER
    </div>
    <ul
      className="brands-list"
    />
  </div>
</div>
`;

exports[`Phones.js file rendering  Phones renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="brands-container"
        >
          <div
            class="header"
          >
            <div
              class="h-c1"
            >
              <img
                alt="logo"
                class="nav-icon"
                src="arrow-left.png"
              />
              <h2
                class="nav-text"
              >
                2022
              </h2>
            </div>
            <div
              class="h-c2"
            >
              <h1
                class="header-title"
              >
                devices by brand
              </h1>
            </div>
            <div
              class="h-c3"
            >
              <img
                alt="logo"
                class="nav-icon"
                src="mic.png"
              />
              <img
                alt="logo"
                class="nav-icon"
                src="gear.png"
              />
            </div>
          </div>
          <div
            class="banner"
          >
            DEVICES BY MAKER
          </div>
          <ul
            class="brands-list"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="brands-container"
      >
        <div
          class="header"
        >
          <div
            class="h-c1"
          >
            <img
              alt="logo"
              class="nav-icon"
              src="arrow-left.png"
            />
            <h2
              class="nav-text"
            >
              2022
            </h2>
          </div>
          <div
            class="h-c2"
          >
            <h1
              class="header-title"
            >
              devices by brand
            </h1>
          </div>
          <div
            class="h-c3"
          >
            <img
              alt="logo"
              class="nav-icon"
              src="mic.png"
            />
            <img
              alt="logo"
              class="nav-icon"
              src="gear.png"
            />
          </div>
        </div>
        <div
          class="banner"
        >
          DEVICES BY MAKER
        </div>
        <ul
          class="brands-list"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
